src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/validator/AppointmentValidator.kt

```diff
@@ -79,35 +92,79 @@ class AppointmentValidator {
+    noSessionReasonLogistics: String?,
     sessionSummary: String?,
     sessionResponse: String?,
     sessionConcerns: String?,
+    notifyProbationPractitioner: Boolean?,
     errors: MutableList<FieldError>,
   ) {
     // if the appointment occurred today or on a date in the future, no attendance validation is required
     if (appointmentTime.isAfter(OffsetDateTime.now().with(LocalTime.MIN))) {
       return
     }
 
+    if (didSessionHappen == null) {
+      errors.add(FieldError(field = "attendanceFeedback.didSessionHappen", error = Code.CANNOT_BE_EMPTY))
+      return
+    }
+
     if (attended == null) {
-      errors.add(FieldError(field = "appointmentAttendance.attended", error = Code.CANNOT_BE_EMPTY))
+      errors.add(FieldError(field = "attendanceFeedback.attended", error = Code.CANNOT_BE_EMPTY))
       return
     }
 
-    when (attended) {
-      NO -> {
-        checkValueNotSupplied(notifyProbationPractitioner, "sessionFeedback.notifyProbationPractitioner", Code.INVALID_VALUE, errors)
-        checkValueNotSupplied(sessionSummary, "sessionFeedback.sessionSummary", Code.INVALID_VALUE, errors)
-        checkValueNotSupplied(sessionResponse, "sessionFeedback.sessionResponse", Code.INVALID_VALUE, errors)
+    if (didSessionHappen) {
+      if (attended != YES) {
+        errors.add(FieldError(field = "attendanceFeedback.attended", error = Code.INVALID_VALUE))
+        return
+      }
+      if (late == null) {
+        errors.add(FieldError(field = "sessionFeedback.late", error = Code.CANNOT_BE_EMPTY))
+        return
+      } else if (late) {
+        checkValueSupplied(lateReason, "sessionFeedback.lateReason", Code.CANNOT_BE_EMPTY, errors)
+      }
+      checkValueSupplied(sessionSummary, "sessionFeedback.sessionSummary", Code.CANNOT_BE_EMPTY, errors)
+      checkValueSupplied(sessionResponse, "sessionFeedback.sessionResponse", Code.CANNOT_BE_EMPTY, errors)
+      checkValueSupplied(notifyProbationPractitioner, "sessionFeedback.notifyProbationPractitioner", Code.CANNOT_BE_EMPTY, errors)
+      if (notifyProbationPractitioner == true) {
+        checkValueSupplied(sessionConcerns, "sessionFeedback.sessionConcerns", Code.CANNOT_BE_EMPTY, errors)
       }
-      else -> { // YES OR LATE
-        checkValueSupplied(notifyProbationPractitioner, "sessionFeedback.notifyProbationPractitioner", Code.CANNOT_BE_EMPTY, errors)
-        checkValueSupplied(sessionSummary, "sessionFeedback.sessionSummary", Code.CANNOT_BE_EMPTY, errors)
-        checkValueSupplied(sessionResponse, "sessionFeedback.sessionResponse", Code.CANNOT_BE_EMPTY, errors)
+    } else {
+      when (attended) {
+        YES -> {
+          when (noSessionReasonType) {
+            null -> errors.add(FieldError(field = "sessionFeedback.noSessionReasonType", error = Code.CANNOT_BE_EMPTY))
+            POP_ACCEPTABLE -> checkValueSupplied(noSessionReasonPopAcceptable, "sessionFeedback.noSessionReasonPopAcceptable", Code.CANNOT_BE_EMPTY, errors)
```

Comment:
I see the value `CODE.CANNOT_BE_EMPTY` in almost all the places set. can this be the default value then?

Reply:
`FieldError() `is used in other places in the app and I'm not sure defaulting the error code to `Code.CANNOT_BE_EMPTY` would be expected?